api/src/services/fields.ts

```diff
@@ -539,7 +550,8 @@ export class FieldsService {
 				}
 			}
 
-			if (hookAdjustedField.meta) {
+			// Only create/update a database record if this is not a system field
+			if (hookAdjustedField.meta && !isSystemField(collection, hookAdjustedField.field)) {
```

Comment:
I'd say for once, the comment is very redundant and maybe it makes more sense to throw an error here instead as right now, while the updating gets skipped, we're still emitting an action event even though nothing changed.

Reply:
The database save gets skipped indeed however we're still emitting an action because the field did get updated (adding/removing the index) even though that is not something we store in our system tables.